src/main/java/database/helper/Validator.java

```diff
@@ -0,0 +1,48 @@
+package database.helper;
+
+import database.exception.EmptyValueException;
+import database.exception.IncorrectPropertyNameException;
+import database.exception.IncorrectValueTypeException;
+import database.exception.NullPropertyNameOrValueException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Map;
+
+public class Validator {
+    private static final Logger logger = LoggerFactory.getLogger(Settings.class);
+
+    public static <V> void validateDatabaseFilters(Field[] declaredFields, Map<String, V> filters)
+            throws NullPropertyNameOrValueException, EmptyValueException, IncorrectPropertyNameException,
+            IncorrectValueTypeException {
+
+        for (Map.Entry<String, V> filter : filters.entrySet()) {
+            String propertyName = filter.getKey();
+            V expectedValue = filter.getValue();
+
+            if (expectedValue == null || propertyName == null) {
+                logger.error("Property name or value is null.");
```

Comment:
It's not an `error`. Error level means that some unexpected problem happened, most likely environment issue or bug. Here we successfully hande cases when user provides incorrect data, so level `info` would be enough. Consider just removing logs from validation.